src/jarabe/gui.py

```diff
@@ -0,0 +1,338 @@
+        self._search_entry.set_text(_(activity_title))
+
+        self._vbox.pack_start(self._hbox, True, True, 0)
+
+        self._buttons_box = Gtk.HButtonBox()
+        self._buttons_box.set_layout(Gtk.ButtonBoxStyle.CENTER)
+        self._buttons_box.set_spacing(style.DEFAULT_SPACING)
+
+        _ok = Gtk.Button()
+        _ok.set_image(Icon(icon_name='dialog-ok'))
+        _ok.set_label(_('Yes'))
+        _ok.connect('clicked', self.__ok_clicked_cb)
+        self._buttons_box.pack_start(_ok, True, True, 0)
+        _ok.show()
+
+        _cancel = Gtk.Button()
+        _cancel.set_image(Icon(icon_name='dialog-cancel'))
+        _cancel.set_label(_('No'))
+        _cancel.connect('clicked', self.__cancel_clicked_cb)
+        self._buttons_box.pack_start(_cancel, True, True, 0)
+        _cancel.show()
+
+        self._vbox.pack_start(self._buttons_box, True, True, 0)
+        self._buttons_box.show()
+        self._search_entry.grab_focus()
+        self.show()
+
+    def _close_cb(self):
+        self.destroy()
+
+    def __cancel_clicked_cb(self, widget):
+        self.destroy()
+
+    def __ok_clicked_cb(self, widget):
+        self.save_screenshot(self._search_entry.get_text())
+        self.destroy()
+
+    def _set_cursor(self, cursor):
+        self.get_window().set_cursor(cursor)
+        Gdk.flush()
+
+    def _set_screensize(self):
+        '''
+        Sets the size of the popup based on
+        the screen resolution.
+        '''
+        width = Gdk.Screen.width() / 4
+        height = Gdk.Screen.height() / 5
+        self.set_size_request(width, height)
+
+    def _show_main_view(self):
+        self._main_view.modify_bg(Gtk.StateType.NORMAL,
+                                  style.COLOR_BLACK.get_gdk_color())
+
+    def __key_press_event_cb(self, window, event):
+        # if the user clicked out of the window - fix SL #3188
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Escape':
+            self.destroy()
+
+        if keyname == 'Return':
+            self.save_screenshot(self._search_entry.get_text())
+            self.destroy()
+
+        if not self.is_active():
+            self.present()
+        return False
+
+    def save_screenshot(self, title):
+        settings = Gio.Settings('org.sugarlabs.user')
+        color = settings.get_string('color')
+
+        jobject = datastore.create()
+        try:
+            jobject.metadata['title'] = title
+            jobject.metadata['keep'] = '0'
+            jobject.metadata['buddies'] = ''
+            jobject.metadata['preview'] = _get_preview_data(
+                self.screenshot_surface)
+            jobject.metadata['icon-color'] = color
+            jobject.metadata['mime_type'] = 'image/png'
+            jobject.file_path = self.file_path
+            datastore.write(jobject, transfer_ownership=True)
+        finally:
+            jobject.destroy()
+            del jobject
+
+
+def generate_thumbnail(screenshot_surface):
```

Comment:
Yes. This function is a modified version of the original ```generate_thumbnail()``` so the code is unique in itself.